### PR TITLE
mistral: Adding support for MLABs as memory

### PR DESCRIPTION
**Pull request by gatecat**
This adds basic support for using MLABs as memory, tested and working with attosoc (where they're used for the register file).

It doesn't support MLAB memories and FFs in the same LAB (I haven't checked if Quartus allows this) nor initialisation - all memories are zero-initialised.

### CODE REVIEW

File: mistral/bitstream.cc [hunk 1]
```diff
@@ -226,11 +226,33 @@ struct MistralBitgen
...
+                cv->bmux_m_set(block_type, pos, CycloneV::MODE, i, CycloneV::RAM);
+                cv->bmux_n_set(block_type, pos, CycloneV::T_FEEDBACK_SEL, i, 1);
+            }
+            cv->bmux_r_set(block_type, pos, CycloneV::LUT_MASK, alm, 0xFFFFFFFFFFFFFFFFULL); // TODO: LUTRAM init
+            cv->bmux_b_set(block_type, pos, CycloneV::BPKREG1, alm, true);
+            cv->bmux_b_set(block_type, pos, CycloneV::TPKREG0, alm, true);
+            cv->bmux_m_set(block_type, pos, CycloneV::MCRG_VOLTAGE, 0, CycloneV::VCCL);
+            cv->bmux_b_set(block_type, pos, CycloneV::RAM_DIS, 0, false);
+            cv->bmux_b_set(block_type, pos, CycloneV::WRITE_EN, 0, true);
+            cv->bmux_n_set(block_type, pos, CycloneV::WRITE_PULSE_LENGTH, 0, 650);
```
Comment (Ravenslofty) [inline mistral/bitstream.cc:245]:
```suggestion
            cv->bmux_n_set(block_type, pos, CycloneV::WRITE_PULSE_LENGTH, 0, 650 /* nanoseconds */);
```

File: mistral/pack.cc [hunk 2]
```diff
@@ -342,12 +342,51 @@ struct MistralPacker
         }
     }
 
+    void constrain_lutram()
+    {
+        // We form clusters based on both read and write address; as both being the same makes it more likely these
+        // cells should be packed together, too This makes things easier for the place,ent legaliser to deal with RAM in
```
Comment (Ravenslofty) [inline mistral/pack.cc:348]:
```suggestion
        // cells should be packed together, too. This makes things easier for the placement legaliser to deal with RAM in
```

**Review by Ravenslofty — COMMENTED** [review]
`Acked-by: Lofty <dan.ravensloft@gmail.com>` with these nits fixed.

**Comment by gatecat** [comment]
@Ravenslofty review comments addressed, I think

**Review by Ravenslofty — APPROVED** [review]

**Pull request merged**